src/compiler/crystal/syntax/location.cr

```diff
@@ -29,6 +29,19 @@ class Crystal::Location
     end
   end
 
+  # Returns the Location whose filename is a String, not a VirtualFile,
+  # traversing virtual file expanded locations leading to the original user source code
+  def user_location
```

Comment:
`original_location` and `user_location` are a bit confusing names and hard to tell apart. Can we find better names?
For example `expanded_location` and `macro_location`? That would describe which location on `VirtualFile` they use.

Reply:
@bcardiff can you show an example where these differ? I intended for original location to be what apparently you want. If not, it's a bug. I don't think we need two methods.

Reply:
```cr
macro foo
  Bar
end

foo
```
Considering `Bar`:

* location: `expanded macro: foo:1:3`: location inside the expanded macro
* original_location: `source.cr:5:1`: where the expanded macro is placed (that's why I'd call this expanded_location)
* user_location: `source.cr:2:3`: location inside the macro source (that' why I'd call this macro_location)

Reply:
The naming proposed by @straight-shoota works for me. In the implementation tool there is a need to track down the location of a method that can result of the expansion of multiple macros. I'm not sure if there are some case unhandled or some macro statements don't track the location fully. But we can review that later.

I'm happy to change original_location to expand_location and introduce macro_location so the feature in this PR moves forwards. Since this are internal compiler things it can be iterated later.

Reply:
Methods renamed